src/plugins/data_explorer/public/components/sidebar/index.tsx

```diff
@@ -111,12 +192,14 @@ export const Sidebar: FC = ({ children }) => {
             onDataSourceSelect={handleSourceSelection}
             selectedSources={selectedSources}
             onGetDataSetError={handleGetDataSetError}
+            dataSourceSelectorConfigs={DATA_SOURCE_SELECTOR_CONFIGS}
             fullWidth
           />
         </EuiSplitPanel.Inner>
         <EuiSplitPanel.Inner paddingSize="none" color="transparent" className="eui-yScroll">
           {children}
         </EuiSplitPanel.Inner>
+        {modalContent}
```

Comment:
Im a little worried that this will cause the sidebar layout to look weird. Can we use this api instead to show the modal? It should clean up the code a lot more.

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/ba298c7daca4f010664fe8a891c427f905115174/src/plugins/discover/public/application/components/top_nav/get_top_nav_links.tsx#L255-L280